garage/experiment/local_tf_runner.py

```diff
@@ -171,8 +172,11 @@ def save_snapshot(self, itr, paths=None):
             paths: Batch of samples after preprocessed.
 
         """
+        assert self.has_setup
+
         logger.log("Saving snapshot...")
-        params = self.algo.get_itr_snapshot(itr, paths)
+        params = self.algo.get_itr_snapshot(itr)
+        params['env'] = self.env
```

Comment:
hmm can you do this in `save_snapshot` instead? that covers the case where an environment has evolving state.

it would be good to test for that case too, since it would be incredibly hard to detect/debug...

Reply:
Are you saying that I should get the snapshot of env from sampler because the copy of env in the runner is not the actual copy used for sampling? 

This leads to another problem: What to save when there are more than one env in the sampler. We can save all envs, but we dont't actually have a mechanism to restore them in sampler. Or we pick one env and save it, and that will be compatible to the current things we save in snapshot.

Reply:
hmm, this presents an interesting problem gap in the interface, but i'm not sure it's one we need to solve right now.

for now i think it's sufficient to make sure there's a copy of `env` in every .pkl file. we can try to handle statefulness later if there's a need.

Reply:
OK. Let's keep an eye on it.